### PR TITLE
dns: send parallel queries to all name servers

### PR DESCRIPTION
**Pull request by ramaraochavali**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/dns/dns.go [hunk 1]
```diff
@@ -298,23 +299,48 @@ func (h *LocalDNSServer) Close() {
...
-		if err == nil {
-			response = cResponse
-			break
-		} else {
-			scope.Infof("upstream failure: %v", err)
-		}
+		upstream := upstream
+		go func() {
+			select {
+			case <-finish:
```
Comment (hzxuzhonghu) [inline pilot/pkg/dns/dns.go:310]: It has no chance to exit when eneters default branch

**Review by howardjohn — COMMENTED** [review]
Any idea how common this is in other dns libraries (core dns, etc)? I recall reading about a few dns servers that do not do this, although I don't recall the motivation

**Comment by ramaraochavali** [comment]
I am actually working on connection reuse instead of connecting on every upstream request based on this plugin https://github.com/coredns/coredns/tree/master/plugin/forward - Then we might not need this . So let us wait on this.

**Comment by ramaraochavali** [comment]
https://github.com/coredns/coredns/blob/master/plugin/forward/forward.go#L99 - coredns actually limits by timeout and does not send to all upstreams if it exceeds the timeout. It does not  send in parallel. I will close this PR and implement connection reuse in a separate PR.

**Pull request closed without merging**